Add option to skip cudagraphing on dynamic shape graphs

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122520

This was requested internally.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D55264528](https://our.internmc.facebook.com/intern/diff/D55264528)

eellison committed:
Add option to skip cudagraphing on dynamic shape graphs

[ghstack-poisoned]

mlazos commented on torch/_inductor/config.py:
```diff
@@ -531,6 +531,10 @@ class triton:
 
+    # Should we skip cudagraphing graphs with dynamic shape inputs
+    # If False, we will re-record a graph for each unique set of shape inputs
+    skip_dynamic_graphs = True
```
nit: should we say cudagraphs_skip_dynamic_graphs

eellison replied:
yea, agreed

mlazos commented on test/inductor/test_cudagraph_trees.py:
```diff
@@ -269,6 +269,19 @@ def foo(x, y):
                 captured_output[0]
             )
 
+        @torch._inductor.config.patch("triton.skip_dynamic_graphs", True)
```
Should you also test that multiple compiles with different shapes still return correct results? or is that the status quo

eellison replied:
already tested: https://github.com/pytorch/pytorch/commit/10cbc9a063d20116ded39fb54a59db9d17e53acf#diff-70ac3047d6e27843ff4f81ee908669839ca9551814ccbd47b2c3354997f87c73R1123-R1180

eellison committed:
Update on "Add option to skip cudagraphing on dynamic shape graphs"


This was requested internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

ezyang commented:
You changed the default, which is fine, but then you need to flip it back for the benchmark

mlazos commented on test/inductor/test_cudagraph_trees.py:
```diff
@@ -269,6 +269,19 @@ def foo(x, y):
                 captured_output[0]
             )
 
+        @torch._inductor.config.patch("triton.cudagraph_skip_dynamic_graphs", True)
```
Should you add an addional test that we properly re-record the shape or is that already covered?

eellison commented:
@ezyang oops good catch

eellison committed:
Update on "Add option to skip cudagraphing on dynamic shape graphs"


This was requested internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55264528).

shunting314 commented on torch/_inductor/compile_fx.py:
```diff
@@ -727,6 +727,31 @@ def fx_codegen_and_compile(
+
+                    if stack_trace := node.meta.get("stack_trace", None):
+                        break
+                disable = "graph with symbolic shapes inputs and config.triton.skip_dynamic_graphs=True."
```
typo on the config name?

eellison committed:
Update on "Add option to skip cudagraphing on dynamic shape graphs"


This was requested internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

Differential Revision: [D55264528](https://our.internmc.facebook.com/intern/diff/D55264528)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55264528).

eellison commented:
@pytorchbot merge
